### ISSUE TITLE
1.20.2

### ISSUE THREAD

**Issue by FallenCrystal** [CatMoe/MoeFilter#62]

**Comment by FallenCrystal** [comment]
Unfortunately. 1.20.2 has too many changes about protocol. This can take a considerable amount of time to complete support. If you want to support 1.20.2, please turn off the limbo feature.

**Comment by FallenCrystal** [comment]
i will tinker with them one by one gradually. (it may be late, dont expect too much.)
Look at dev branch: [dev-1-20-2](https://github.com/CatMoe/MoeFilter/tree/dev-1-20-2)

commits until now:
 - https://github.com/CatMoe/MoeFilter/commit/a3d46c70d6d378b3a8ec0763fd98faf80573b97d
 - https://github.com/CatMoe/MoeFilter/commit/8f031177fc0671078ba0bfda084c52a657910174
 - https://github.com/CatMoe/MoeFilter/commit/9edb9aafb41be435279151417c8d577bf09559f9
 - https://github.com/CatMoe/MoeFilter/commit/1601f6c2cad7e4da7b4e961f7311370ec4c210ec

to-do list:
  - [x] Add 1.20.2 version in [version list](https://github.com/CatMoe/MoeFilter/commit/a3d46c70d6d378b3a8ec0763fd98faf80573b97d#diff-4e3aba36409bf28147a943048d210e8c9a2f2b23066e095438c012ec183656a0)
  - [x] Added new packets and updated packet IDs for 1.20.2
  - [x] Prepare the "configuration" process separately for 1.20.2. And parse and respond to packets correctly.
  - [x] Fix and improved nbt tag write (Especially 1.20.2 and chunk packets.)
  - [x] Can successfully join the game.
  - [x] AntiBot checks supports for 1.20.2.

**Comment by FallenCrystal** [comment]
log (joining limbo, 1.20.2 fabric)
```raw
[Render thread/ERROR]: Error executing task on Client
java.lang.IllegalStateException: Missing registry: ResourceKey[minecraft:root / minecraft:dimension_type]
	at net.minecraft.class_5455.method_30534(class_5455.java:26) ~[client-intermediary.jar:?]
	at java.util.Optional.orElseThrow(Optional.java:403) ~[?:?]
	at net.minecraft.class_5455.method_30530(class_5455.java:26) ~[client-intermediary.jar:?]
	at net.minecraft.class_634.method_11120(class_634.java:391) ~[client-intermediary.jar:?]
	at net.minecraft.class_2678.method_11567(class_2678.java:57) ~[client-intermediary.jar:?]
	at net.minecraft.class_2678.method_11054(class_2678.java:14) ~[client-intermediary.jar:?]
	at net.minecraft.class_2600.method_11072(class_2600.java:23) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.method_18859(class_1255.java:156) ~[client-intermediary.jar:?]
	at net.minecraft.class_4093.method_18859(class_4093.java:23) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.method_16075(class_1255.java:130) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.method_5383(class_1255.java:115) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1523(class_310.java:1231) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1514(class_310.java:856) ~[client-intermediary.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:253) ~[1.20.2.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:468) ~[fabric-loader-0.14.22.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) ~[fabric-loader-0.14.22.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) ~[fabric-loader-0.14.22.jar:?]
[Render thread/ERROR]: Error executing task on Client
java.lang.NullPointerException: Cannot invoke "net.minecraft.class_1657.method_18798()" because "$$1" is null
	at net.minecraft.class_634.method_11157(class_634.java:607) ~[client-intermediary.jar:?]
	at net.minecraft.class_2708.method_11740(class_2708.java:51) ~[client-intermediary.jar:?]
	at net.minecraft.class_2708.method_11054(class_2708.java:9) ~[client-intermediary.jar:?]
	at net.minecraft.class_2600.method_11072(class_2600.java:23) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.method_18859(class_1255.java:156) ~[client-intermediary.jar:?]
	at net.minecraft.class_4093.method_18859(class_4093.java:23) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.method_16075(class_1255.java:130) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.method_5383(class_1255.java:115) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1523(class_310.java:1231) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1514(class_310.java:856) ~[client-intermediary.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:253) ~[1.20.2.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:468) ~[fabric-loader-0.14.22.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) ~[fabric-loader-0.14.22.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) ~[fabric-loader-0.14.22.jar:?]
[Render thread/ERROR]: Error executing task on Client
java.lang.NullPointerException: Cannot invoke "net.minecraft.class_638.method_27873(net.minecraft.class_2338, float)" because "this.field_45588.field_1687" is null
	at net.minecraft.class_634.method_11142(class_634.java:949) ~[client-intermediary.jar:?]
	at net.minecraft.class_2759.method_11869(class_2759.java:29) ~[client-intermediary.jar:?]
	at net.minecraft.class_2759.method_11054(class_2759.java:7) ~[client-intermediary.jar:?]
	at net.minecraft.class_2600.method_11072(class_2600.java:23) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.method_18859(class_1255.java:156) ~[client-intermediary.jar:?]
	at net.minecraft.class_4093.method_18859(class_4093.java:23) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.method_16075(class_1255.java:130) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.method_5383(class_1255.java:115) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1523(class_310.java:1231) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1514(class_310.java:856) ~[client-intermediary.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:253) ~[1.20.2.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:468) ~[fabric-loader-0.14.22.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) ~[fabric-loader-0.14.22.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) ~[fabric-loader-0.14.22.jar:?]
[Render thread/ERROR]: Error executing task on Client
java.lang.NullPointerException: Cannot invoke "net.minecraft.class_638.method_29089(long)" because "this.field_45588.field_1687" is null
	at net.minecraft.class_634.method_11079(class_634.java:941) ~[client-intermediary.jar:?]
	at net.minecraft.class_2761.method_11872(class_2761.java:38) ~[client-intermediary.jar:?]
	at net.minecraft.class_2761.method_11054(class_2761.java:6) ~[client-intermediary.jar:?]
	at net.minecraft.class_2600.method_11072(class_2600.java:23) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.method_18859(class_1255.java:156) ~[client-intermediary.jar:?]
	at net.minecraft.class_4093.method_18859(class_4093.java:23) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.method_16075(class_1255.java:130) ~[client-intermediary.jar:?]
	at net.minecraft.class_1255.method_5383(class_1255.java:115) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1523(class_310.java:1231) ~[client-intermediary.jar:?]
	at net.minecraft.class_310.method_1514(class_310.java:856) ~[client-intermediary.jar:?]
	at net.minecraft.client.main.Main.main(Main.java:253) ~[1.20.2.jar:?]
	at net.fabricmc.loader.impl.game.minecraft.MinecraftGameProvider.launch(MinecraftGameProvider.java:468) ~[fabric-loader-0.14.22.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:74) ~[fabric-loader-0.14.22.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) ~[fabric-loader-0.14.22.jar:?]
```
